app_dart/lib/src/request_handlers/update_status_cache.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:collection/collection.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+import 'package:neat_cache/neat_cache.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../model/appengine/agent.dart';
+import '../model/appengine/commit.dart';
+import '../request_handling/body.dart';
+import '../request_handling/request_handler.dart';
+import '../service/build_status_provider.dart';
+import '../service/datastore.dart';
+
+/// Cache writer for /api/public/get-status.
+/// 
+/// Queries datastore to get the latest information on the commits
+/// to flutter/flutter and the agents in the devicelab.
+@immutable
+class UpdateStatusCache extends RequestHandler<Body> {
```

Comment:
We shouldn't need this handler -- if the caching handler populates the cache entries whenever it has to delegate to the underlying handler, then you get free cache population by virtue of wrapping a handler in a caching handler.

Reply:
Currently it does not do that. However, this is a much more elegant solution and I have implemented it that way.

Reply:
That's how you're gonna create the stream from the list, but you could create a `Body.forBytes(Uint8List bytes)` factory constructor in `Body` to abstract this away, so your call here would become `return Body.forBytes(cachedResponse)`.  (`cachedResponse` will be a `Uint8List` after you make the other suggested changes)